src/backend/executor/nodeShareInputScan.c

```diff
@@ -925,9 +923,12 @@ writer_wait_for_acks(ShareInput_Lk_Context *pctxt, int share_id, int xslice)
 			int save_errno = errno;
 			elog(LOG, "SISC WRITER (shareid=%d, slice=%d): notify still wait for an answer, errno %d",
 					share_id, currentSliceId, save_errno);
-			/*if error(except EINTR) happens in select, we just return to avoid endless loop*/
-			if(errno != EINTR){
-				return;
+			if(save_errno == EBADF)
+			{
+				/* The file description is invalid, maybe this FD has been already closed by writer in some cases
+				 * we need to break here to avoid endless loop and continue to run CHECK_FOR_INTERRUPTS.
+				 */
+				break;
```

Comment:
@ictmalili ,  already printed log there, so need to to add code.
@paul-guo- I just process the error code as doc described, if there are some bug is OS, then we should check log when the hung problem occur again. At that time, we can find out the root cause. We just need to make sure not to hung by hawq code bug here. Thanks.